packages/server/modules/user-emails/migrations/20240703084247_user-emails.ts

```diff
@@ -0,0 +1,25 @@
+import { Knex } from 'knex'
+
+const USER_EMAILS_TABLE_NAME = 'user_emails'
+
+export async function up(knex: Knex): Promise<void> {
+  await knex.schema.createTable(USER_EMAILS_TABLE_NAME, (table) => {
+    table.string('id').notNullable().primary()
```

Comment:
@iainsproat Can you elaborate on both points? I don't fully understand what you mean in the 1st scenario, and the 2nd point is a new one to me - why not?

Reply:
Can't you do that now already? If you're first to register an e-mail you're first and no one can do that afterwards, regardless of whether its verified or not.

Reply:
Not sure if that's true, if we still mostly rely on users.email and just use this for extra emails. I'm also concerned about opening up a whole new can of worms of multiple people having the same email attached and all of the validation and syncing needed to manage all of that.

That said - I suppose this can be merged and adjusted afterwards when there's an actual usage of this. I don't really get the point of PRs this small when the change doesn't really make sense in isolation, I would've put this together with the logic that actually uses the table.

One thing that I think we should still add is at least a unique index on (userId, email)

Reply:
1. the idea is to drop the `email` col in the `users` table and only use this one. For this reason we need the `primary` column here
2. makes sense to me, we can make the association userId + email as primary key, I don't particularly like the idea of having the email as the PK alone

Reply:
I think it is better if we connect later on this @fabis94 @iainsproat